robotoff/insights/dataclass.py

```diff
@@ -99,7 +99,7 @@ def from_raw_insight(
             data=insight.data,
             value_tag=insight.value_tag,
             value=insight.value,
-            automatic_processing=insight.automatic_processing or False,
+            automatic_processing=insight.automatic_processing,
```

Comment:
[not in scope for this PR, more of a design discussion]: 

It seems to me that the root of the issue is the fact that we're allowing a nullable primitive type when we don't seem to care about the nullability.

I.e. here we're storing this bool as an `Optional[Bool]`, but the database field is not optional and defaults to `False`. 
From a design perspective, it would make sense to have it optional if during the insight processing we cared to differentiate between `True`, `False` and `I don't know`.

Looking at the current usage, this doesn't seem to be the case as we always want to treat `None` as `False`, but because we allow both values it can easily result in bugs like these if you're not super careful.

Does it make sense to try to remove the Optional qualifier from primitives like these (I think 'latent' is in a similar situation) going forward if we're ever refactoring this code?

Reply:
Correct me if I'm wrong, but we do want to differentiate between `True`, `False`, and `I don't know (for now)`. The purpose was to allow insight extractor (OCR-based mostly) to specify the insights for which we have enough confidence to be processed automatically (`True`), the insights for which it is the opposite (False), and to fall-back to default behaviour for this insight type (`None`).

Reply:
I see.

I was interpreting it as the insight extractor is the only place where we can make a decision as to whether something is automatically processable since it understands the thresholds for the specific insights. Hence, once the `RawInsight` is constructed nothing else can affect the processability of the insight. But if we want to be able to arbitrarily flip the default then it makes sense I guess.